vault/request_handling.go

```diff
@@ -518,6 +531,11 @@ func (c *Core) handleLoginRequest(req *logical.Request) (retResp *logical.Respon
 		auth.Accessor = te.Accessor
 		auth.Policies = te.Policies
 
+		// Set auth.TTL if not set, used below in RegisterAuth
+		if auth.TTL == 0 {
+			auth.TTL = tokenTTL
```

Comment:
This can be set to be the period's value if that's present. Not sure if we want that, or have it set to DefaultLeaseTTL.

Reply:
tokenTTL should already be the period's value from above if it's present. Actually, I think what needs to happen here is that in all cases we need to set auth.TTL to the tokenTTL value. That will ensure that the expiration manager always uses the correct value.

Reply:
Yea, I'll do `auth.TTL = te.TTL` since these two should be set to the same value.